packages/core/contracts/gnosis-zodiac/implementation/OptimisticGovernor.sol

```diff
@@ -215,39 +249,38 @@ contract OptimisticGovernor is Module, Lockable {
-        optimisticOracle.requestPrice(identifier, time, ancillaryData, collateral, 0);
-        uint256 totalBond = optimisticOracle.setBond(identifier, time, ancillaryData, bondAmount);
-        optimisticOracle.setCustomLiveness(identifier, time, ancillaryData, liveness);
+        // Check that the proposal is not already mapped to an assertionId, i.e., is not a duplicate.
+        require(proposalHashes[proposalHash] == bytes32(0), "Duplicate proposals not allowed");
 
-        // Get the bond from the proposer and approve the bond and final fee to be used by the oracle.
-        // This will fail if the proposer has not granted the OptimisticGovernor contract an allowance
+        // Get the bond from the proposer and approve the required bond to be used by the Optimistic Asserter.
+        // This will fail if the proposer has not granted the Optimistic Governor contract an allowance
         // of the collateral token equal to or greater than the totalBond.
+        uint256 totalBond = getProposalBond();
         collateral.safeTransferFrom(msg.sender, address(this), totalBond);
-        collateral.safeIncreaseAllowance(address(optimisticOracle), totalBond);
-
-        optimisticOracle.proposePriceFor(
-            msg.sender,
-            address(this),
-            identifier,
-            time,
-            ancillaryData,
-            PROPOSAL_VALID_RESPONSE
-        );
+        collateral.safeIncreaseAllowance(address(optimisticAsserter), totalBond);
+
+        // Assert that the proposal is correct at the Optimistic Asserter.
+        bytes32 assertionId =
+            optimisticAsserter.assertTruth(
+                claim, // claim containing proposalHash, explanation and rules.
+                proposer, // asserter will receive back bond if the assertion is correct.
+                address(this), // callbackRecipient is set to this contract for automated proposal deletion on disputes.
+                escalationManager, // escalationManager (if set) used for whitelisting proposers / disputers.
+                liveness, // liveness in seconds.
+                collateral, // currency in which the bond is denominated.
+                totalBond, // bond amount used to assert proposal.
+                identifier, // identifier used to determine if the claim is correct at DVM.
+                bytes32(0) // domainId is not set.
+            );
 
-        uint256 challengeWindowEnds = time + liveness;
+        // Maps the proposal hash to the returned assertionId and vice versa.
+        proposalHashes[proposalHash] = assertionId;
+        assertionIds[assertionId] = proposalHash;
 
-        emit TransactionsProposed(proposer, time, proposal, proposalHash, _explanation, challengeWindowEnds);
+        emit TransactionsProposed(proposer, time, assertionId, proposal, proposalHash, _explanation, time + liveness);
```

Comment:
Do we also need emitting rules as part of TransactionsProposed? This would though increase LOC due to splitting this across multiple lines and Optimistic Asserter emits claim that will have rules appended.

Reply:
it would still be part of the same event, right? I think it's ok to emit it as well. ideally events are as expressive as possible! extra LOC for more expressiveness in events seems worth it to me.